skimage/measure/_regionprops.py

```diff
@@ -398,15 +398,17 @@ def local_centroid(self):
 
     @property
     def max_intensity(self):
-        return np.max(self.intensity_image[self.image], axis=0)
+        return np.max(self.intensity_image[self.image], axis=0)\
+                 .astype(np.double)
```

Comment:
Is this cast necessary?

Reply:
Unfortunately yes for now, since leaving `dtype` free breaks the consistency of the `dtype` with `COL_DTYPES` and makes the test `test_column_dtypes_correct` fail. I guess one might need a bit more of code dev to remove this cast.